internal/host/hostutil/host_utils.go

```diff
@@ -354,9 +354,17 @@ func GetIgnitionEndpoint(cluster *common.Cluster, host *models.Host) (string, er
 		poolName = host.MachineConfigPoolName
 	}
 
+	protocol := "http"
+	port := constants.InsecureMCSPort
+	if common.HasCACertInIgnition(host.IgnitionConfigOverrides) {
```

Comment:
What does a user have to do to make this true? If we're moving from requiring the user to change a value in one CR to just making them change a value in another one this doesn't seem like much of an improvement.

Reply:
This should work out of the box  for ZTP,  BMH controller set the CA as IgnitionOverrides.
see https://github.com/openshift/assisted-service/blob/c84135a7decfce9ecdeb322f49ebbb02a45991ad/internal/controller/controllers/bmh_agent_controller.go#L1370